R/coord-cartesian-.r

```diff
@@ -153,10 +214,41 @@ view_scales_from_scale <- function(scale, coord_limits = NULL, expand = TRUE) {
   view_scales
 }
 
-draw_view_scale_axis <- function(view_scale, axis_position, theme) {
-  if(is.null(view_scale) || view_scale$is_empty()) {
+panel_guide_label <- function(guides, position, default_label) {
+  guides <- guides_filter_by_position(guides, position)
+
+  if (length(guides) == 0 || is.null(guides[[1]]$title)) {
+    default_label
+  } else {
+    guides[[1]]$title %|W|% default_label
+  }
+}
+
+panel_guides_grob <- function(guides, position, theme) {
+  guides <- guides_filter_by_position(guides, position)
+  grobs <- lapply(guides, guide_gengrob, theme)
+
+  if (length(grobs) == 0) {
     return(zeroGrob())
+  } else if (length(grobs) == 1) {
+    grobs[[1]]
+  } else {
+    warning(
```

Comment:
There is something weird going on with the logic here... If I modify your own examples I get:
```r
p <- ggplot(mpg, aes(cty * 100, hwy * 100)) +
    geom_point() +
    facet_wrap(vars(class))

# axis guides can be customized in the scale_* functions or
# using guides()
p + scale_x_continuous(guide = guide_axis(position = 'top'))
#> Warning messages:
#> 1: In panel_guides_grob(panel_params$guides, position = "top", theme = theme) :
#>   More than one position guide found at `position = "top". Only showing the first guide.
#> 2: In panel_guides_grob(panel_params$guides, position = "top", theme = theme) :
#>   More than one position guide found at `position = "top". Only showing the first guide.
#> 3: In panel_guides_grob(panel_params$guides, position = "top", theme = theme) :
#>   More than one position guide found at `position = "top". Only showing the first guide.
#> 4: In panel_guides_grob(panel_params$guides, position = "top", theme = theme) :
#>   More than one position guide found at `position = "top". Only showing the first guide.
#> 5: In panel_guides_grob(panel_params$guides, position = "top", theme = theme) :
#>   More than one position guide found at `position = "top". Only showing the first guide.
#> 6: In panel_guides_grob(panel_params$guides, position = "top", theme = theme) :
#>   More than one position guide found at `position = "top". Only showing the first guide.
#> 7: In panel_guides_grob(panel_params$guides, position = "top", theme = theme) :
#>   More than one position guide found at `position = "top". Only showing the first guide.
```

Reply:
Can you provide an example of how it doesn't play nicely with `labs()`